### PR TITLE
TokenRules contract deployment

### PR DESCRIPTION
**Pull request by gulshanvasnani**
PR contains following :-
1. TokenRules contract abi and bin.
2. TokenRules contract deployer script.
3. Unit test case to verify the deployment of token rules.
4. Web3WalletHelper.js to setup the accounts and required configuration files.
5. Integrated with deployer service.

Fixes :- #51

### CODE REVIEW

**Comment by abhayks1** [comment]
Let's integrate it with Deployer.js common library.

File: lib/setup/TokenRules.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    oThis.auxiliaryWeb3 = auxiliaryWeb3;
+    oThis.address = address;
+    oThis.abiBinProvider = new AbiBinProvider();
+  }
+
+  /**
+   * Performs deployment of TokenRules contract.
+   *
+   * @param _organization Organization which holds all the keys needed to administer the economy.
+   * @param _token EIP20Token address.
```
Comment (abhayks1) [inline lib/setup/TokenRules.js]: _token EIP20 token contract address deployed for an economy.

File: lib/setup/TokenRules.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    txReceipt = await new Deployer(params).deploy();
+
+    return txReceipt;
+  }
+
+  /**
+   * Performs deployment of TokenRules contract.
+   *
+   * @param _organization Organization which holds all the keys needed to administer the economy.
+   * @param _token EIP20Token address.
```
Comment (abhayks1) [inline lib/setup/TokenRules.js]: _token EIP20 token contract address deployed for an economy.

For these contract variables, documentation can be copied from contracts

File: utils/deployContract.js [hunk 3]
```diff
@@ -1,56 +1,29 @@
 'use strict';
 
-const DeployContract = function(params) {
+const DeployContract = function (params) {
   const oThis = this;
```
Comment (abhayks1) [inline utils/deployContract.js]: space is not needed after function. review and correct formatting all over the file.

File: lib/setup/TokenRules.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+const AbiBinProvider = require('../AbiBinProvider');
+const Deployer = require('../../utils/deployContract');
+
+const ContractName = 'TokenRules';
+
+/**
+ * Helper method which performs deployment of Token Rules contract.
+ */
+class TokenRules {
+  constructor(auxiliaryWeb3, address) {
```
Comment (abhayks1) [inline lib/setup/TokenRules.js]: documentation for constructor is missing.
Why do we need address parameter here in constructor?

File: lib/setup/TokenRules.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+   * Performs deployment of TokenRules contract.
+   *
+   * @param _organization Organization which holds all the keys needed to administer the economy.
+   * @param _token EIP20Token address.
+   * @param txOptions Tx options.
+   */
+  async deploy(_organization, _token, txOptions) {
+    const oThis = this;
+
+    let tx = oThis._deployRawTx(_organization, _token, txOptions);
```
Comment (abhayks1) [inline lib/setup/TokenRules.js]: lets rename tx => txObject everywhere.

File: package.json [hunk 6]
```diff
@@ -33,9 +33,13 @@
     "node-cmd": "3.0.0",
     "path": "0.12.7",
     "shelljs": "0.8.1",
-    "web3": "1.0.0-beta.34"
+    "web3": "1.0.0-beta.37",
+    "path": "0.12.7"
```
Comment (abhayks1) [inline package.json]: path package is being added 2 times here.

File: test/integration/setup/TokenRules.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+  web3WalletHelper = new Web3WalletHelper(auxiliaryWeb3),
+  abiBinProvider = new AbiBinProvider(),
+  assert = chai.assert;
+
+describe('TokenRules', async function() {
+
+  let options = {
+    from: config.deployerAddress,
+    gasPrice: config.gasPrice,
+    gas: '8000000'
```
Comment (abhayks1) [inline test/integration/setup/TokenRules.js]: let's use the gas value from configReader. Avoid hardcoding values in test files

File: utils/deployContract.js [hunk 8]
```diff
@@ -1,56 +1,29 @@
...
-      transactionHash = null,
-      receipt = null;
-
+    let receipt = null,
+      transactionHash = null;
     console.log('Deploying contract ' + oThis.contractName);
-    let instance = await tx
-      .send(txOptions)
-      .on('receipt', function(value) {
+    let instance = await oThis.tx
```
Comment (abhayks1) [inline utils/deployContract.js]: oThis.tx => oThis.txObject

File: utils/deployContract.js [hunk 9]
```diff
@@ -1,56 +1,29 @@
...
+      .on('receipt', function (value) {
         receipt = value;
       })
-      .on('transactionHash', function(value) {
+      .on('transactionHash', function (value) {
         console.log('transaction hash: ' + value);
         transactionHash = value;
       })
-      .on('error', function(error) {
+      .on('error', function (error) {
```
Comment (abhayks1) [inline utils/deployContract.js]: Avoid using space after function

File: lib/setup/TokenRules.js (hunk 4 above)
Comment (gulshanvasnani) [reply lib/setup/TokenRules.js]: Going forward we would have **registerRule** method.So,for constructing contract instance we would need the tokenrules contract address. This is the reason **address** parameter is needed.

**Pull request merged**